packages/mason_cli/lib/src/commands/make.dart

```diff
@@ -232,6 +284,60 @@ class _MakeCommand extends MasonCommand {
       return value;
     }
   }
+
+  /// Starts watching for changes.
+  ///
+  /// Should only be called when the flag `--watch` is specified.
+  ///
+  /// When watching, any changes done within the brick's directory will trigger
+  /// a new make command [run] (with the same arguments as the first [run]).
+  ///
+  /// This method does nothing when:
+  /// - The brick is not local.
+  /// - The brick does not have a path.
+  /// - The command is already watching for changes.
+  ///
+  /// See also:
+  ///
+  /// * [DirectoryWatcher], watcher used to trigger new makes.
+  Future<void> _watch() async {
+    if (!localBricks.contains(_brick) || _brick.path == null || _isWatching) {
+      return;
+    }
+    _isWatching = true;
+
+    final brickDirectoryPath = p.dirname(_brick.path!);
+
+    final boldBrickName = styleBold.wrap(_brick.name);
+    logger.info(
+      '👀 Watching for $boldBrickName changes in $brickDirectoryPath',
+    );
+
+    final directoryWatcher = DirectoryWatcher(
+      brickDirectoryPath,
+      pollingDelay: pollingDelay,
+    );
+
+    final watchSubscription = directoryWatcher.events.listen(
+      (event) async {
+        logger.info(
+          '\n👀 Detected changes, remaking $boldBrickName brick',
+        );
+        // TODO(alestiago): Consider using onError and onData instead.
+        await run();
+      },
+    );
+
+    // TODO(alestiago): Consider adding prompt to confirm exit.
```

Comment:
@felangel would we like to add a confirmation prompt to end the watch? If so, I think we could consider adding so in a different Pull Request. Wdyt?